src/Storages/MergeTree/StorageFromMergeTreeDataPart.h

```diff
@@ -59,6 +59,21 @@ class StorageFromMergeTreeDataPart final : public IStorage
         return std::make_shared<StorageSnapshot>(*this, metadata_snapshot, std::move(object_columns));
     }
 
+    StorageSnapshotPtr getStorageSnapshotForQuery(
+        const StorageMetadataPtr & metadata_snapshot, const ASTPtr & /*query*/, ContextPtr /*query_context*/) const override
+    {
+        const auto & storage_columns = metadata_snapshot->getColumns();
+        if (!hasDynamicSubcolumns(storage_columns))
+            return std::make_shared<StorageSnapshot>(*this, metadata_snapshot);
+
+        auto data_parts = storage.getDataPartsVectorForInternalUsage();
+
+        auto object_columns = getConcreteObjectColumns(
+            data_parts.begin(), data_parts.end(), storage_columns, [](const auto & part) -> const auto & { return part->getColumns(); });
+
+        return std::make_shared<StorageSnapshot>(*this, metadata_snapshot, std::move(object_columns));
+    }
```

Comment:
Please, leave only `getStorageSnapshot`. `getStorageSnapshotForQuery` calls `getStorageSnapshot` by default and for this storage overriding of `getStorageSnapshotForQuery` is not needed.

Reply:
Fixed. https://github.com/ClickHouse/ClickHouse/pull/54052/commits/7dda1646fd8c507ee32b53e35db3d7493150bd6c